soroban-env-common/src/status.rs

```diff
@@ -366,7 +366,7 @@ mod tests {
         ];
 
         let pairs: Vec<_> = xdr_vals
-            .into_iter()
+            .iter()
             .map(|xdr_val| {
```

Comment:
I think if you wanted to keep the `into_iter()` you'd remove the `xdr_val.clone()`. I think clippy is removing into_iter() because it sees you only need a borrow because of the clone.

Reply:
agreed

Reply:
The clone is necessary in this case though right? Maybe I'm missing your point.

Reply:
into_iter() moves the items through the iterator, instead of hte iterator providing references to items that need cloning.

Reply:
For example, see this code that compiles, not the types on the left side of each:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7a10092653bee5ec2bd91d12f86bcc0c
```rust
fn main() {
    let list = vec!["1".to_string(),"2".to_string(),"3".to_string(),"4".to_string()];
    let refs: Vec<&String> = list.iter().map(|i| i).collect::<Vec<_>>();
    let cloned_in_map: Vec<String> = list.iter().map(|i| i.clone()).collect::<Vec<_>>();
    let cloned: Vec<String> = list.iter().cloned().collect::<Vec<_>>();
    let moved: Vec<String> = list.into_iter().collect::<Vec<_>>();
}
```

Reply:
(I'm going to resolve this since it's not critical)